tools/cp-utility/src/main.rs

```diff
@@ -42,21 +44,32 @@ struct CopyOperation {
 
 /// Parse command line arguments and transform into `CopyOperation`
 fn parse_args(args: Vec<&str>) -> io::Result<CopyOperation> {
-    if !(args.len() == 3 || args.len() == 4 && args[1].eq("-a")) {
+    if !(args.len() == 3
+        || args.len() == 4 && args[1].eq("-a")
+        || args.len() == 4 && args[1].eq("-r"))
+    {
```

Comment:
Is the associativity correct here or are more parens needed to make sure the logic is correct?  I don't see a test for this function, maybe that's needed.  Could probably also be simplified:

```suggestion
    if !(args.len() == 3
        || (args.len() == 4 && (args[1].eq("-a") || args[1].eq("-r")))
    {
```

Reply:
yes yes, simplified the associativity and added parser_correct unit test